lib/charms/operator_libs_linux/v2/snap.py

```diff
@@ -584,13 +584,16 @@ def ensure(
                     "Installing snap %s, revision %s, tracking %s", self._name, revision, channel
                 )
                 self._install(channel, cohort, revision)
-            else:
+                logger.info("The snap installation completed successfully")
+            elif (revision is None) or (revision != self._revision):
```

Comment:
In my experience using the extra parens is atypical here (and unnecessary):

```suggestion
            elif revision is None or revision != self._revision:
```